pkg/domain/domain.go

```diff
@@ -2250,6 +2250,10 @@ func (do *Domain) UpdateTableStatsLoop(ctx, initStatsCtx sessionctx.Context) err
 	// This is because the updated worker's primary responsibilities are to update the change delta and handle DDL operations.
 	// These tasks do not interfere with or depend on the initialization process.
 	do.wg.Run(func() { do.updateStatsWorker(ctx, owner) }, "updateStatsWorker")
+	do.wg.Run(func() {
+		<-do.StatsHandle().InitStatsDone
```

Comment:
`select` with `ctx` to avoid dead wait?

Reply:
I have removed the logical that wait for init-stats.